lib/repl.js

```diff
@@ -296,15 +322,60 @@ function REPLServer(prompt,
           return;
         }
       }
-    }
 
-    // After executing the current expression, store the values of RegExp
-    // predefined properties back in `savedRegExMatches`
-    for (var idx = 1; idx < savedRegExMatches.length; idx += 1) {
-      savedRegExMatches[idx] = RegExp[`$${idx}`];
+      if (awaitPromise && !err) {
+        let sigintListener;
+        pause();
+        const promise = self.breakEvalOnSigint ? Promise.race([
+          result,
+          new Promise((resolve, reject) => {
+            sigintListener = () => {
+              reject(new Error('Script execution interrupted.'));
+            };
+            priorizedSigintQueue.push(sigintListener);
+          })
+        ]) : result;
```

Comment:
I'd argue this reads better as an if statement:
```js
let promise = result;
if (!self.breakEvalOnSigint) {
  const alt = new Promise((resolve, reject) => {
    // ...
  });
  promise = Promise.race([promise, alt]);
}

Reply:
Done.